okhttp/src/main/java/okhttp3/internal/connection/RealConnection.java

```diff
@@ -288,6 +288,7 @@ private void connectTls(ConnectionSpecSelector connectionSpecSelector) throws IO
       // Create the wrapper over the connected socket.
       sslSocket = (SSLSocket) sslSocketFactory.createSocket(
           rawSocket, address.url().host(), address.url().port(), true /* autoClose */);
+      socket = sslSocket;
```

Comment:
motivation? if we keep this you’ll wanna remove the corresponding assignment on line 325.

as-is I think I like the previous behavior better. We don’t promote the SSL socket until we’ve handshaked. But that might leak now?